File: api/post-to-notion.py
import requests
import os
import json

NOTION_TOKEN = os.environ["NOTION_TOKEN"]
NOTION_VERSION = os.environ.get("NOTION_VERSION", "2022-06-28")

def handler(request):
    try:
        # Read incoming JSON body
        body = json.loads(request.body.decode())

        # Send request to Notion
        r = requests.post(
            "https://api.notion.com/v1/pages",
            headers={
                "Authorization": f"Bearer {NOTION_TOKEN}",
                "Content-Type": "application/json",
                "Notion-Version": NOTION_VERSION
            },
            data=json.dumps(body)
        )

        return r.json(), r.status_code

    except Exception as e:
        return {"error": str(e)}, 500

File: post-to-notion.js
export default async function handler(req, res) {
  if (req.method !== "POST") {
    return res.status(405).json({ error: "Method not allowed" });
  }

  const NOTION_TOKEN = process.env.NOTION_TOKEN;
  const NOTION_VERSION = "2022-06-28";

  if (!NOTION_TOKEN) {
    return res.status(500).json({ error: "NOTION_TOKEN is not set" });
  }

  try {
    const body = req.body;

    const response = await fetch("https://api.notion.com/v1/pages", {
      method: "POST",
      headers: {
        Authorization: `Bearer ${NOTION_TOKEN}`,
        "Content-Type": "application/json",
        "Notion-Version": NOTION_VERSION
      },
      body: JSON.stringify(body)
    });

    const data = await response.json();
    return res.status(response.status).json(data);
  } catch (err) {
    return res.status(500).json({ error: err.message });
  }
}
